crowdsourcehinter/static/js/src/crowdsourcehinter.js

```diff
@@ -213,7 +213,7 @@ function CrowdsourceHinter(runtime, element, data){
     function submitNewHint(){ return function(submitHintButtonHTML){
     //add the newly created hint to the hinter's pool of hints
         if($('.csh_student_text_input', element).val().length > 0){
-            var studentAnswer = unescape(submitHintButtonHTML.currentTarget.attributes['answer'].value);
+            var studentAnswer = decodeURI(submitHintButtonHTML.currentTarget.attributes['answer'].value);
```

Comment:
It'd be helpful to know why this is necessary. I can't follow this code, and you won't be able to either six months later. 

Generally, for data transformations like this, you want something like: 

     # This is data submitted from a form GET request. It is URL-encoded, and hashed. For example, "http://localhost/events/current/event=something_or_other"
    # We'd like to get just the URL parameters, event=something_or_other
    params = request.split('/')[-1]
    # Now, we'd like to make a dictionary of values, {"event":"something_or_other"}
    param_dict = [p.split("=") for p in params.split("&")]
    # Now, we'd like to fix up URL encoding. For example, instead of {"event":"John+Smith+submits"}, we'd like {"event": "John Smith submits"}